app/components/SignInSignOut.jsx

```diff
@@ -0,0 +1,44 @@
+import React, { Component, PropTypes } from 'react';
+import Spinner from 'react-mdl/lib/Spinner';
+import Constants from '../constants';
+import AuthActions from '../actions/auth';
+
+export default class SignInSignOut extends Component {
+  render() {
+    const { dispatch, auth } = this.props;
+    let content;
+
+    switch (auth.state) {
```

Comment:
Yes, I noticed this behaviour :)

I was not sure whether to already drop it :P

I was also not able to reproduce the spinner case (authentication too fast)